engine/src/vault/key_store.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::vault::{
+    crypto_box::{BoxProvider, Key, NCKey},
+    VaultId,
+};
+use std::collections::HashMap;
+
+/// The [`KeyStore`] keeps a map of [`VaultId`] -> [Vec<u8>] representing
+/// encrypted [`Key<P>`] using the `master_key`.
+/// `master_key` is stored in a non-contiguous data structure [`NCKey<P>`]
+/// for more security
+pub struct KeyStore<P: BoxProvider> {
+    store: HashMap<VaultId, Vec<u8>>,
+    master_key: NCKey<P>,
+}
+
+impl<P: BoxProvider> KeyStore<P> {
+    #![allow(dead_code)]
```

Comment:
Again, I think `KeyStore` should also be public outside of the crate so we can use it in `iota_stronghold`. In that case we also don't need the `#![allow(dead_code)]`.

Reply:
Resolved with #324.